sync_settings/libs/gist.py

```diff
@@ -63,14 +65,22 @@ def make_uri(endpoint=''):
     def create(self, data):
         if not isinstance(data, dict) or not len(data):
             raise ValueError('Gist can`t be created without data')
-        return self.__do_request('post', self.make_uri(), data=json.dumps(data)).json()
+        return self.__do_request(
+             'post',
+             self.make_uri(),
+             data=sublime.encode_value(data, True).encode('utf-8').decode('latin-1'),
+         ).json()
 
     @auth
     @with_gid
     def update(self, gid, data):
         if not isinstance(data, dict) or not len(data):
             raise ValueError('Gist can`t be updated without data')
-        return self.__do_request('patch', self.make_uri(gid), data=json.dumps(data)).json()
+        return self.__do_request(
+             'patch',
+             self.make_uri(gid),
+             data=sublime.encode_value(data, True).encode('utf-8').decode('latin-1'),
```

Comment:
same here

Reply:
@mfuentesg turns out this isn't a proper solution as the issue causing this is when reading binary files such as a image file. Which `json.dumps` also handles incorrectly.
I am going to revert most changes here except that which solves the trailing comma and open a new PR for a solution for handling binary data.

Reply:
reverted changes due to larger PR coming